command/app.go

```diff
@@ -122,3 +114,28 @@ func Main(ctx context.Context, args []string) error {
 
 	return app.RunContext(ctx, args)
 }
+
+// s3opts returns new S3Options object by extracting
+// its fields from the provided context.
+func s3opts(c *cli.Context, isSrc bool) storage.S3Options {
```

Comment:
We can remove the boolean parameter by introducing another function:

```go
func s3opts(c *cli.Context) storage.S3Options {
	region := c.String("source-region")
	if region == "" {
		region = c.String("default-source-region")
	}
	return storage.S3Options{
		MaxRetries:  c.Int("retry-count"),
		Endpoint:    c.String("endpoint-url"),
		NoVerifySSL: c.Bool("no-verify-ssl"),
		Region:      region,
	}
}

func dstS3opts(c *cli.Context) storage.S3Options {
	dstRegion := c.String("region")
	if dstRegion == "" {
		dstRegion = c.String("default-region")
	}
	return storage.S3Options{
		MaxRetries:  c.Int("retry-count"),
		Endpoint:    c.String("endpoint-url"),
		NoVerifySSL: c.Bool("no-verify-ssl"),
		Region:      dstRegion,
	}
}
```

This way it will be more readable on the usage side (with the cost of a little code duplication)
```go
opts := s3opts(c, false)
// vs
opts := dstS3opts(c)
```
I prefer readability.